### DISCUSSION TITLE
Problème de homing

### DISCUSSION THREAD

**Discussion by GillotSebastien** [fra589/grbl-Mega-5X#351] [General]
Bonjour,

j'ai terminé la construction de ma 3ème CNC. Je suis pour celle ci, parti sur le modèle LOWRIDER V3 car elle mesure 1.5mx2.0m. J'ai pris le firmware GrblMega5x car je veux commencer à tester l'usinage avec un petit tour sur la table de fraisage.

Ma CNC est équipée de 5 moteurs : 1X, 2Y, 2Z, 1 plus tard pour le A (rotation).
J'utilise une Mega 2560 sans shield RAMPS car j'ai 5 drivers TB6600.

J'ai configuré mes axes comme cela :

`#define N_AXIS 6     `
`#define N_AXIS_LINEAR 5`
`#define AXIS_1 0       `
`#define AXIS_1_NAME 'X' `
`#define AXIS_2 1`
`#define AXIS_2_NAME 'Y'`
`#define AXIS_3 2`
`#define AXIS_3_NAME 'Z'`
`#if N_AXIS > 3`
`  #define AXIS_4 3`
`  #define AXIS_4_NAME 'Y' `
`#endif`
`#if N_AXIS > 4`
 ` #define AXIS_5 4`
`  #define AXIS_5_NAME 'Z' `
`#endif`
`#if N_AXIS > 5`
`  #define AXIS_6 5`
`  #define AXIS_6_NAME 'A' `
`#endif`

Ma machine possède 5 interrupteurs de fin de course en NC comme cela :

1X,2Y,2Z

J'ai procédé à une config homing comme ca :

`#elif N_AXIS == 6`
`  #define HOMING_CYCLE_0 ((1<<AXIS_3)|(1<<AXIS_5)) `
`  #define HOMING_CYCLE_1 (1<<AXIS_1)  `
`  #define HOMING_CYCLE_2 ((1<<AXIS_2)|(1<<AXIS_4)) `

Mon problème est que lorsque je démarre un homing, les 2 moteurs en Z bougent d'un demi millimètre et j'ai une erreur 1 qui apparait en désactivant les enable des drivers.

J'ai essayé de modifier :
$5 en 0 ou 1
$21 en 0 ou 1
$24,$25,$26,$27 en plusieurs valeurs
Rien ne fonctionne.
J'ai donc pensé à des interférences sur les switches. J'ai monté tous mes switches sur relais indépendants avec des optocoupleurs et rien ne fonctionne. J'ai inversé physiquement le NC en NO avec les relais, cela ne fonctionne pas non plus.

Je n'ai plus d'idée sur ce que je peux faire.
Si je passe à 0 les paramètres $20 et $22, tous mes mouvements de broche manuelle fonctionne parfaitement. C'est simplement une histoire de homing et de soft-limit.
J'ai câblé les drivers et les switches selon le schéma : https://github.com/fra589/grbl-Mega-5X/wiki/grbl-Mega-5X-pinout tout en axe min. Je n'ai rien sur l'axe max.

Si quelqu'un peut m'aider.  Merci beaucoup.

Sébastien

**Comment by fra589** [comment]
Bonjour Sébastien,

Interrupteurs fin de courses NC je ne suis pas sûr ?
C'est des fin de courses "Normalement Ouverts" (Le contact électrique se fait lorsque le fin de course est activé) 
ou bien "Normalement Fermés" (le contact électrique se coupe lorsque le fin de course est activé) ?
Ensuite, de quels cotés sont les fins de course ? Coté X, Y & Z mini ou maxi ?

Si c'est des contacte Normalement Fermés, il faut paramétrer $5 = 1 et relier les pins de fin de courses non utilisés à la masse.

Par exemple, si tu as branché tes fin de courses de manière standard pour une CNC (coté max des axes), il faudra shunter les pins des contacts fin de course min à la masse.

Si tu as branché tes fin de course coté min, il faudra shunter les fin de courses max à la masse. En plus, il faudra configurer le paramètre $23 ([Homing direction invert Mask](https://github.com/fra589/grbl-Mega-5X/wiki/Grbl-Configuration#23---homing-dir-invert-mask)).

Enfin, l'erreur 1 me parait pas normal. L'erreur 1, c'est une erreur d'analyse de la commande GCode (il manque un lettre dans la ligne de GCode.

Quelle commande utilises-tu pour le Homing ?
As tu essayé de faire les axes séparément ? 
`$HZ`
`$HX`
`$HY`

@++,
Gauthier.

P.S. au passage, envoi la sortie des commandes `$$` et `$I`
